web-ncer/app/services/SampleRequestService.scala

```diff
@@ -0,0 +1,298 @@
+    fieldNames: Seq[String],
+    selectedIds: Seq[BSONObjectID]
+  ): Future[String] = {
+    val SEPARATOR = "\t"
+    val dsa = dsaf(dataSetId).getOrElse(throw new IllegalArgumentException(s"Dataset '$dataSetId' does not exist."))
+    val fieldCriteria = if (fieldNames.nonEmpty) Vector(FieldIdentity.name #-> fieldNames) else Nil
+    val selectCriteria = if (selectedIds.nonEmpty) Vector(JsObjectIdentity.name #-> selectedIds) else Nil
+    for {
+      fields <- dsa.fieldRepo.find(fieldCriteria)
+      valueCriteria <- FieldUtil.toDataSetCriteria(dsa.fieldRepo, conditions)
+      items <- dsa.dataSetRepo.find(valueCriteria ++ selectCriteria)
+    } yield {
+      val header = fields.map(_.name)
+      val csv = new StringBuilder("")
+      csv ++= header.mkString(SEPARATOR)
+      csv += '\n'
+      items foreach { item =>
+        val row = header map { header =>
+          (item \ header).getOrElse(JsNull).toString
+        }
+        csv ++= row.mkString(SEPARATOR)
+        csv += '\n'
+      }
+      csv.toString
+    }
+  }
+
+  /**
+   * Creates an application in REMS
+   *
+   * @param csv A String representing a valid CSV file used as an application attachment
+   * @param catalogueItemId A REMS catalogue item to create an application for
+   * @param catalogueFormId The REMS form id used by the catalogue item
+   * @param user The Ada user who makes the request
+   * @return The URL pointing to the newly created application
+   */
+  def sendToRems(
+    csv: String,
+    catalogueItemId: Int,
+    catalogueFormId: Int,
+    user: User
+  ): Future[String] = {
+    for {
+      applicationId <- createApplication(catalogueItemId, user)
+      attachmentId <- addAttachment(applicationId, csv, user)
+      _ <- saveDraft(applicationId, attachmentId, catalogueFormId, user)
+    } yield {
+      s"$remsUrl/application/$applicationId"
+    }
+  }
+
+  /**
+   * Get available REMS catalogue items
+   *
+   * @return A sequence of catalogue items
+   */
+  def getCatalogueItems: Future[Seq[CatalogueItem]] =
+    for {
+      res <- ws.url(remsUrl + "/api/catalogue").withHeaders(
+        "x-rems-user-id" -> remsServiceUser,
+        "x-rems-api-key" -> remsMasterApiKey
+      ).get()
+    } yield {
+      if (res.status != 200) throw new AdaException("Failed to retrieve catalogue items from REMS. Reason: " + res.body)
+      res.json.as[Seq[JsObject]] map { catalogueItemJson =>
+        CatalogueItem(
+          (catalogueItemJson \ "id").as[Int],
+          (catalogueItemJson \ "localizations" \ "en" \ "title").as[String],
+          (catalogueItemJson \ "formid").as[Int]
+        )
+      }
+    }
+
+  /**
+   * Build object containing all data needed to render the sample request submission form
+   *
+   * @param dataSetId The data set id used
+   * @param request The current request
+   * @return An ActionFormViewData object
+   */
+  def getActionFormViewData(
+    dataSetId: String
+  )(
+    implicit request: AuthenticatedRequest[_]
```

Comment:
similar to `FilterCondition`... services should not depend on any UI stuff such as `AuthorizedRequest`. This should go to a controller